File: 백준/Silver/14469. 소가 길을 건너간 이유 3/소가 길을 건너간 이유 3.py
import sys
input = sys.stdin.readline

n = int(input())

c = []

for _ in range(n):
    # a, b = map(int, input().split())
    # l = list(map(int,))
    # c.append([a, b])
    a, k = map(int, input().split())
    c.append([a, k])

c.sort(key=lambda x: x[0])

ans = 0

for i in range(n):
    if ans >= c[i][0]:
        ans += c[i][1]
    else:
        ans = ans + (c[i][0] - ans + c[i][1])
    # print('i = ', i, ans)


print(ans)

File: 백준/Silver/14469. 소가 길을 건너간 이유 3/소가 길을 건너간 이유 3.js
const readline = require('readline');
 
// 인터페이스 객체를 만들자.
const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});

let n;
let arr = []

rl.on('line',line=>{
    if(!n){
        n = parseInt(line)
    }else{
        arr.push(line.split(' ').map(Number));
    }
    
}).on('close',()=>{
    // console.log(arr)
    arr.sort((a,b)=>{
        if(a[0] === b[0]){
            return a[1] - b[1]
        }
        return a[0] - b[0]
    })
    let cnt = 0;
    for(let [start,end] of arr){
        if(cnt <= start){
            cnt = start + end;
            // cnt += start+end;
        }else{
            cnt += end;
        }
    }
    console.log(cnt)
})